docs/source/user-guide/integrations/h2o.rst

```diff
@@ -0,0 +1,157 @@
+scikit-learn API that deepchecks expects.
+
+We have created a simple wrapper for our H2O model, that implements the required API for deepchecks according the the
+:doc: `</user-guide/tabular/supported_models>` guide. This wrapper is available from
+`deepchecks.tabular.integrations.h2o`.
+
+Generally the wrapper model will contain 2 functions in
+case of a classification problem: the ``predict`` and the ``predict_proba`` functions that will be called by
+deepchecks. In addition, it is also possible to specify the feature importances of the model. Read more about
+feature importance handling in the :doc: `</user-guide/tabular/feature_importance>` guide.
+
+.. code:: ipython3
+
+    from deepchecks.tabular.integrations.h2o import H2OWrapper
+    h2o_dc_gbm = H2OWrapper(gbm)
+
+Now we can run deepchecks! We’ll start by defining the deepchecks
+Dataset objects and comparing the two models.
+
+.. code:: ipython3
+
+    from deepchecks.tabular import Dataset
+
+.. code:: ipython3
+
+    train_ds = Dataset(train_df, label=label_col, features=feature_cols, cat_features=adult._CAT_FEATURES)
+    test_ds = Dataset(test_df, label=label_col, features=feature_cols, cat_features=adult._CAT_FEATURES)
+
+Comparing model performance
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: ipython3
+
+    from deepchecks.tabular.checks import MultiModelPerformanceReport
+
+.. code:: ipython3
+
+    MultiModelPerformanceReport().run(train_ds, test_ds, [h2o_dc_gbm, skmodel])
+
+
+We can see that from an overall performance standpoint (considering the
+f1 metric), the H2O model achieves superior performance compared to the
+sklearn model, especially in the smaller class.
+
+Further model performance evaluation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We next wish to get a more in depth evaluation of the H2O model. For example,
+we can use the SegmentPerformance check to compare the performance of
+the two models across segments of the sex and relationship status
+features.
+
+.. code:: ipython3
+
+    from deepchecks.tabular.checks import SegmentPerformance
+    # For a reason that has to do with the way h2o handles multiprocessing, we must change the parallel_backend from
+    # loky to threading for all the of the checks to be able to compute while the h2o model is inferring.
+    from sklearn.utils import parallel_backend
+    with parallel_backend('threading', n_jobs=2):
+        result_h2o = SegmentPerformance(feature_1='sex', feature_2='relationship').run(test_ds, h2o_dc_gbm)
+        
+    result_skl = SegmentPerformance(feature_1='sex', feature_2='relationship').run(test_ds, skmodel)
+
+
+.. parsed-literal::
+
+    Calculating permutation feature importance. Expected to finish in 56 seconds
+
+
+.. code:: ipython3
+
+    result_h2o
+
+
+.. code:: ipython3
+
+    result_skl
+
+
+We see that the scikit-learn model, apart from being worse overall, is
+specifically worse (a drop in accuracy of 0.6) at predicting the income
+of females with a relationship status of “Others”.
```

Comment:
Finishing off the notebook - currently feels a bit "cut in the middle"
consider with a general sentence or two?  (e.g. summarizing the high-level steps required, the fact that now they can run any check/suite etc.)